src/Module/Conversation/Channel.php

```diff
@@ -0,0 +1,438 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Conversation;
+
+use Friendica\App;
+use Friendica\App\Mode;
+use Friendica\BaseModule;
+use Friendica\Content\BoundariesPager;
+use Friendica\Content\Conversation;
+use Friendica\Content\Feature;
+use Friendica\Content\Nav;
+use Friendica\Content\Text\HTML;
+use Friendica\Content\Widget;
+use Friendica\Content\Widget\TrendingTags;
+use Friendica\Core\Cache\Capability\ICanCache;
+use Friendica\Core\Cache\Enum\Duration;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig\Capability\IManagePersonalConfigValues;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\Contact;
+use Friendica\Model\Post;
+use Friendica\Model\User;
+use Friendica\Module\Security\Login;
+use Friendica\Network\HTTPException;
+use Friendica\Core\Session\Model\UserSession;
+use Friendica\Database\Database;
+use Friendica\Module\Response;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class Channel extends BaseModule
+{
+	const WHATSHOT  = 'whatshot';
+	const FORYOU    = 'foryou';
+	const FOLLOWERS = 'followers';
+	const IMAGE     = 'image';
+	const VIDEO     = 'video';
+	const AUDIO     = 'audio';
+
+	protected static $content;
+	protected static $accountTypeString;
+	protected static $accountType;
+	protected static $itemsPerPage;
+	protected static $min_id;
+	protected static $max_id;
+	protected static $item_id;
+
+	/** @var UserSession */
+	private $session;
+	/** @var ICanCache */
+	private $cache;
+	/** @var IManageConfigValues The config */
+	private $config;
+	/** @var SystemMessages */
+	private $systemMessages;
+	/** @var App\Page */
+	private $page;
+	/** @var Conversation */
+	private $conversation;
```

Comment:
See comment in `Update\Channel.php`.
```suggestion
	protected $conversation;
```

Reply:
Changed